test/activerecord-clean-db-structure/primary_key_test.rb

```diff
@@ -61,37 +68,95 @@ class PrimaryKeyTest < Minitest::Spec
         --
 
         ALTER TABLE ONLY public.storage_tables_blobs
           ADD CONSTRAINT storage_tables_blobs_pkey PRIMARY KEY (checksum, partition_key);
 
+        --
+        -- Name: storage_tables_blobs_partition_0 storage_tables_blobs_partition_0_pkey; Type: CONSTRAINT; Schema: public; Owner: -
+        --
+
+        ALTER TABLE ONLY public.storage_tables_blobs_partition_0
+        ADD CONSTRAINT storage_tables_blobs_partition_0_pkey PRIMARY KEY (checksum, partition_key);
+
+        --
+        -- Name: index_storage_tables_blobs_on_checksum; Type: INDEX; Schema: public; Owner: -
+        --
+
+        CREATE INDEX index_storage_tables_blobs_on_checksum ON ONLY public.storage_tables_blobs USING btree (checksum) WHERE (attachments_count = 0);
+
+        --
+        -- Name: storage_tables_blobs_partition_0_checksum_idx; Type: INDEX; Schema: public; Owner: -
+        --
+
+        CREATE INDEX storage_tables_blobs_partition_0_checksum_idx ON public.storage_tables_blobs_partition_0 USING btree (checksum) WHERE (attachments_count = 0);
+
+        -- Name: storage_tables_blobs_partition_0 immutable_blob_key_and_checksum; Type: TRIGGER; Schema: public; Owner: -
+        --
+
+        CREATE TRIGGER immutable_blob_key_and_checksum BEFORE UPDATE ON public.storage_tables_blobs_partition_0 FOR EACH ROW EXECUTE FUNCTION public.immutable_blob_key_and_checksum();
+
         --
         -- PostgreSQL database dump complete
         --
       SQL
 
-      assert_equal <<~SQL, described_class.new(dump).run
+      assert_equal <<~SQL, described_class.new(dump, options).run
 
         -- Name: storage_tables_blobs; Type: TABLE
 
         CREATE TABLE public.storage_tables_blobs (
-          partition_key character(1) NOT NULL,
-          checksum character(85) NOT NULL,
-          attachments_count_modified timestamp(6) with time zone DEFAULT CURRENT_TIMESTAMP NOT NULL,
+          PRIMARY KEY (checksum, partition_key),
           attachments_count integer DEFAULT 0 NOT NULL,
+          attachments_count_modified timestamp(6) with time zone DEFAULT CURRENT_TIMESTAMP NOT NULL,
           byte_size bigint NOT NULL,
+          checksum character(85) NOT NULL,
           content_type character varying,
-          metadata jsonb
+          metadata jsonb,
+          partition_key character(1) NOT NULL
         )
         PARTITION BY LIST (partition_key);
 
-        -- Name: storage_tables_blobs storage_tables_blobs_pkey; Type: CONSTRAINT
+        CREATE INDEX index_storage_tables_blobs_on_checksum ON public.storage_tables_blobs USING btree (checksum) WHERE (attachments_count = 0);
 
-        ALTER TABLE ONLY public.storage_tables_blobs
-          ADD CONSTRAINT storage_tables_blobs_pkey PRIMARY KEY (checksum, partition_key);
+        -- Name: storage_tables_blobs_partition_0; Type: TABLE
+
+        CREATE TABLE public.storage_tables_blobs_partition_0 (
+          PRIMARY KEY (checksum, partition_key),
```

Comment:
Werkt dit? Ik zou verwachten dat de columns eerst aanwezig moeten zijn voordat de primary key gezet kan worden.

Reply:
Nee dat had niet zo moeten gebeuren, had getest met deze versie van gem in debiteurenbeheer en daar deed hij het anders. Heb nu zo aangepast dat hij met de sortering ze net als de constraints achteraan plaats.